src/go/rpk/pkg/cli/cluster/storage/mount.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package storage
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/redpanda-data/common-go/rpadmin"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/adminapi"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+)
+
+func newMountCommand(fs afero.Fs, p *config.Params) *cobra.Command {
+	var to string
+
+	cmd := &cobra.Command{
+		Use:   "mount [TOPIC]",
+		Short: "Mount a topic to the Redpanda Cluster from Tiered Storage",
+		Long: `Mount a topic from Tiered Storage to the Redpanda cluster.
+
+This command mounts a topic in the Redpanda Cluster using log segments stored
+in Tiered Storage.
+
+Requirements:
+- Tiered storage must be enabled.
+- The topic must have a minimum of three partitions.
+- Log segments for the topic must be available in Tiered Storage.
+- A topic with the same name must not already exist in the cluster.`,
+		Example: `
+Mounts topic my-typic from Tiered Storage to the cluster in the my-namespace
+	rpk cluster storage mount my-topic
+
+Mount topic my-topic from Tiered Storage to the cluster in the my-namespace 
+with my-new-topic as the new topic name
+	rpk cluster storage mount my-namespace/my-topic --to my-namespace/my-new-topic
```

Comment:
Are other namespaces supported? It appears that at least the Admin API call only supports `kafka`